test/Microsoft.AspNet.Mvc.ModelBinding.Test/BodyBindingAndValidationIntegrationTest.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.ComponentModel.DataAnnotations;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.Framework.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Test
+{
+    public class BodyValidationIntegrationTests
+    {
+        private class Person
+        {
+            [FromBody]
+            [Required]
+            public Address Address { get; set; }
+        }
+
+        private class Address
+        {
+            public string Street { get; set; }
+        }
+
+        [Fact]
+        public async Task BodyBoundOnProperty_RequiredOnProperty_AddsModelStateError()
```

Comment:
name is hard to parse.  think you mean something like `FromBodyOnProperty_RequiredOnProperty_...` or `PropertyWithFromBodyAndRequired_...`